taskset.go

```diff
@@ -220,14 +221,69 @@ func (ts *TaskSet) FilterUnorganised() {
 	}
 }
 
-func (ts *TaskSet) MustGetByID(id int) Task {
+func (ts *TaskSet) MustGetByID(id int) (Task, error) {
 	if ts.tasksByID[id] == nil {
-		ExitFail("No open task with ID %v exists.", id)
+		return Task{}, fmt.Errorf("No open task with ID %v exists.", id)
 	}
 
-	return *ts.tasksByID[id]
+	return *ts.tasksByID[id], nil
 }
 
+func (ts *TaskSet) SearchForUUID(uuid string) (string, error) {
+	var uuids []string
+	for _, task := range ts.tasks {
+		if strings.HasPrefix(task.UUID, uuid) {
+			uuids = append(uuids, task.UUID)
+		}
+	}
+	if len(uuids) == 0 { // No matches
+		return "", fmt.Errorf("No task matching incomplete UUID: %s exists.\n", uuid)
+
+	} else if len(uuids) > 1 { // More than one UUID found that matches string
+
+		// Print UUIDs and Summary and prompt user to enter more specific ID.
+		// Probably better to either filter ts or create new taskset with
+		// uuids then can use DisplayByNext
+		fmt.Printf("\nUUID \t\t\t\t\tSummary\n")
+		for _, uuid := range uuids {
+			task, err := ts.MustGetByUUID(uuid)
+			if err != nil { //This should never happen
+				ExitFail("Unable to retrieve UUID: %s: ", uuid, err)
+			}
+			fmt.Printf("%s \t%s\n", task.UUID, task.Summary)
+		}
+		fmt.Println()
+		return "", fmt.Errorf("Multiple Tasks Match UUID: %s \nProvide more unique Partial UUID\n", uuid)
+	}
+	return uuids[0], nil
+}
+
+func (ts *TaskSet) MustGetByUUID(uuid string) (Task, error) {
+	if !IsValidUUID4String(uuid) {
+		uuid, err := ts.SearchForUUID(uuid)
+		if err != nil {
+			return Task{}, err
+		}
+		return *ts.tasksByUUID[uuid], nil
+	} else if ts.tasksByUUID[uuid] == nil {
+		return Task{}, fmt.Errorf("No task with UUID %v exists in %v.", uuid, ts)
+	} else {
+		return *ts.tasksByUUID[uuid], nil
+	}
+}
```

Comment:
After reviewing this yesterday. I'm not sure that I'm happy with how I've implemented this. Line 263 calls SearchForUUID, but I don't think that function is necessary. I could use the Filter method of TaskSet. For example, by adding the following to the `func (task *Task) MatchesFilter(cmdLine CmdLine)` I can filter on UUIDs. 

```go 
if cmdLine.UUID != "" && !strings.HasPrefix(task.UUID, uuid) {
    return false
}
```

I wonder if it wouldn't be better to pass a CmdLine to MustGetByUUID() and do something like this...
```go
if !IsValidUUID4String(cmdLine.UUID){
    if !IsValidPartialUUID4String(cmdLine.UUID) {
        // return Task{}, error
    } else {
        ts.Filter(cmdLine)
        tasks := ts.Tasks()
        if len(tasks) == 0 {
            // return Task{}, error
        } else if len(tasks) > 1 {
            ts.ShowMatchingUUIDs  // function to be created in display.go
            // return Task{}, error
        } else {
            return tasks[0], nil
        }
    }
...
```
There may be a tradeoff in performance... Putting the UUID test at the bottom of MatchesFilter() means that when searching for a resolved UUID, every resolved task will first have to check for matching ID, tags, antitag, antiprojects, projects, priority and Summary/Note text before checkign for matching UUID. Putting it more towards the top just means that when this is called for TaskSets loaded from NON_RESOLVED_STATUSES, they will be checking for cmdLine.UUIDs when there will be none. Maybe that's ok. 

 Using ts.Filter(cmdLine) has the added benefit of being able to filter on things other than just UUID (tags, projects, summary/notes text, etc). I can see this coming in handy when working with resolved tasks in the thousands. I have already seen some of my resolved UUIDs match up 6-7 digits with just a few hundred tasks. Just adding a tag or a project name to the cmdline may remove the need for disambiguation most of the time. 

 I also ran into the issue within SearchForUUID() where I couldn't create the disambiguation display code in display.go easily because I wasn't working on the TaskSet ts itself. Most of the display functions rely on the task.filtered parameter.